oct/tournament/serializers.py

```diff
@@ -150,10 +155,21 @@ class TournamentMatchSerializer:
         'referee',
         'streamer',
         'commentator1',
-        'commentator2'
+        'commentator2',
+        'time_str',
+        'winner',
+        'has_started',
+        'progress'
     ]
 
-
+    def _transform(self, obj, fields, exclude, include):
+        data = super()._transform(obj, fields, exclude, include)
+        # Adding custom method fields
+        if 'has_started' in fields:
+            data['has_started'] = obj.get_has_started()
+        if 'progress' in fields:
+            data['progress'] = obj.get_progress()
+        return data
```

Comment:
instead of this, turn the functions into properties and put it in `fields`

Reply:
the functions are properties on the model, without the _transform override they just return null in the jsonresponse when they're in the fields

Reply:
if they were properties, you wouldn't be calling them + checking the model they're not properties

Reply:
I'm an idiot I was thinking of something else, sorry. Fixed.